client/landing/stepper/hooks/use-get-site-suggestions-query.ts

```diff
@@ -0,0 +1,32 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+
+type SuggestionsResponse =
+	| {
+			success: true;
+			suggestions: { title: string }[];
+	  }
+	| {
+			success: false;
+	  };
+
+export const getSiteSuggestions = (): Promise< SuggestionsResponse > =>
+	wpcom.req.get( {
+		method: 'GET',
+		apiNamespace: 'wpcom/v2',
+		path: '/site-suggestions',
+	} );
+
+export const useGetSiteSuggestionsQuery = ( {
+	enabled,
+	onSuccess,
+}: {
+	enabled: boolean;
+	onSuccess?: ( response: SuggestionsResponse ) => void;
+} ) =>
+	useQuery( {
+		queryKey: [ 'site-suggestions' ],
```

Comment:
I see that the new `useGetSiteSuggestionsQuery` hook is just a refactor so it has the same behaviour as before.

I'm wondering how each instance of the hook could be made independent? Now that it's shared, if `useGetSiteSuggestionsQuery` was used in two different components, would we want them to share the same suggestion? I feel we'd want them each to have unique suggestions.

One way might to give each instance it's own unique query key? i.e.

```ts
import { useInstanceId } from '@wordpress/components';

// ...

const instanceId = useInstanceId();

useQuery( {
	queryKey: [ 'site-suggestions', instanceId ],
	// ...
} );
```

Maybe we could also return the result of `useQuery` from the custom hook. Just for completeness.